### PR TITLE
Add additional api calls

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
- calculateQuery
    - getAggregateTrackingForMailing
    - getMailingTemplate
    
    I have given these their own classes as I thought that displaying parameters
    as properties makes them transparent & allows for more to be added
    in a clear way

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
I just rebased this. It also adds support for downloading the sftp file in the main class now

**Comment by mrmarkfrench** [comment]
Hey, sorry I've been slow to respond on this. I'm in the middle of moving, and packing up my whole life has not seemed to make it simpler and more efficient, just chaotic.

**Comment by eileenmcnaughton** [comment]
Thanks for responding :-)

In the meantime I'm extending your files in my own package. I'm also looking at switching from curl to Guzzle. I believe it's fairly easy to intercept guzzle for the purposes of unit tests - which I can't quite figure out how to add without the change.

File: src/SilverpopConnector/SilverpopXmlConnector.php [hunk 1]
```diff
@@ -996,4 +1072,31 @@ protected function post($params, $pathExtension='/XMLAPI', $urlParams='') {
...
+	 * Sample code:
+	 *
+	 * $status = $this->silverPop->getJobStatus($this->getJobStatus();
+	 *   if ($status === 'COMPLETE')) {
+	 *     $file = $result->downloadFile();
+	 *   }
+	 *
+	 * @return bool
+	 */
+	public function downloadFile($fileName, $destination) {
```
Comment (mrmarkfrench) [inline src/SilverpopConnector/SilverpopXmlConnector.php]: This public method should be sorted alphabetically with the other public methods in the top area of the file.

File: src/SilverpopConnector/Xml/XmlAction.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * Created by IntelliJ IDEA.
+ * User: emcnaughton
+ * Date: 5/10/17
+ * Time: 8:23 AM
+ */
+class XmlAction {
+
+  public function __construct($params) {
```
Comment (mrmarkfrench) [inline src/SilverpopConnector/Xml/XmlAction.php]: Shouldn't this class be declared as `abstract`?

In fact, given that all of the child classes need a `formatResult()` method, adding that as an abstract method here would help to make the expected interface for the various child classes more clear.

**Comment by mrmarkfrench** [comment]
Thanks for taking the time to contribute! I appreciate the effort at making the connector better.

**Comment by eileenmcnaughton** [comment]
I've just pushed a commit addressing tabs & ordering of functions & taking on board your suggestions about the abstract function.

With regards to the class structure. I do think that structure better supports flexibility going forwards and I'm happy to work on converting some of the other functions - although not all in one hit.

The tricky thing is the parameters  - e.g it might have to look like

	public function exportList(
		$listId,
		$startDate = null,
		$endDate   = null,
		$type      = self::EXPORT_TYPE_ALL,
		$format    = self::EXPORT_FORMAT_CSV,
		$columns   = array()) {

if (is_array($listId) ) {
// call the class based file
} 

else 

do some wrangling first

**Comment by mrmarkfrench** [comment]
Maybe, but it might also be easier to do the wrangling in the class, and leave the methods in the existing location as just a pass-through. It seems like you've got a pretty clear idea what you want to do, at least in the big picture, so I'll be interested to see what you come up with for some of those individual questions. Thanks again for the contribution!

**Comment by eileenmcnaughton** [comment]
Just noting that #27 converts one class to be in line with this structure, although the focus of that PR is largely on adding unit testing, to allow such refactoring to be done

**Pull request merged**